subsys/net/lib/aws_fota/include/aws_fota_json.h

```diff
@@ -27,10 +27,36 @@ extern "C" {
  */
 #define STATUS_MAX_LEN (12)
 
-int aws_fota_parse_notify_next_document(char *job_document,
-		u32_t payload_len, char *job_id_buf, char *hostname_buf,
-		char *file_path_buf);
+/**
+ * @brief Parse a given AWS IoT DescribeJobExecution response JSON object.
+ *	  More information on this object can be found at https://docs.aws.amazon.com/iot/latest/developerguide/jobs-api.html#mqtt-describejobexecution
+ *
+ * @param[in] job_document JSON formated string of an AWS Job execution
+ * @param[in] payload_len The length of the provided string.
+ * @param[out] job_id_buf Output buffer for the Job ID string read from the JSON
+ * @param[out] hostname_buf Output buffer for the hostname read from the JSON
+ * @param[out] file_path_buf Output buffer for the file path read from the JSON
+ * @param[out] version_number Job document version number.
+ *
+ * @return 1 if the Job Execution object is empty, 0 if parsed correctly,
```

Comment:
Maybe a nit, but this scheme seems weird to me: Return 1 if 0 objects were found, return 0 if 1 object was found and parsed successfully. Consider switching them?

Reply:
Looking at the function, it seems like "0 if parsed correctly" is wrong since `ret` is returned and BIT(2) has to be set.

Reply:
yeah the documentation here is wrong. 
I'll take your suggestion and switch them.

Reply:
Fixed.